python/lsst/daf/butler/core/datasets/type.py

```diff
@@ -472,6 +474,94 @@ def _lookupNames(self) -> Tuple[LookupKey, ...]:
 
         return lookups + self.storageClass._lookupNames()
 
+    def to_simple(self, minimal: bool = False) -> Union[Dict, str]:
+        """Convert this class to a simple python type suitable for
+        serialization.
+
+        Parameters
+        ----------
+        minimal : `bool`, optional
+            Use minimal serialization. Requires Registry to convert
+            back to a full type.
+
+        Returns
+        -------
+        simple : `dict` or `str`
+            The object converted to a dictionary or a simple string.
+        """
+        if minimal:
+            # Only needs the name.
+            return self.name
+
+        # Convert to a dict form
+        as_dict = {"name": self.name,
+                   "storageClass": self._storageClassName,
+                   "isCalibration": self._isCalibration,
+                   "dimensions": self.dimensions.to_simple(),
+                   }
+
+        if self._parentStorageClassName is not None:
+            as_dict["parentStorageClass"] = self._parentStorageClassName
+        return as_dict
+
+    @classmethod
+    def from_simple(cls, simple: Union[Dict, str],
+                    universe: Optional[DimensionUniverse] = None,
+                    registry: Optional[Registry] = None) -> DatasetType:
+        """Construct a new object from the data returned from the `to_simple`
+        method.
+
+        Parameters
+        ----------
+        simple : `dict` of [`str`, `Any`] or `str`
+            The value returned by `to_simple()`.
+        universe : `DimensionUniverse`
+            The special graph of all known dimensions of which this graph will
+            be a subset. Can be `None` if a registry is provided.
+        registry : `lsst.daf.butler.Registry`, optional
+            Registry to use to convert simple name of a DatasetType to
+            a full `DatasetType`. Can be `None` if a full description of
+            the type is provided along with a universe.
+
+        Returns
+        -------
+        datasetType : `DatasetType`
+            Newly-constructed object.
+        """
+        if isinstance(simple, str):
+            if registry is None:
+                raise ValueError(f"Unable to convert a DatasetType name '{simple}' to DatasetType"
+                                 " without a Registry")
+            datasetTypes = list(registry.queryDatasetTypes(simple))
```

Comment:
I suspect this is going to be a fair bit slower than `Registry.getDatasetType`, and it should be impossible to get more than one result anyway; there's a unique constraint in the database, and the pattern-matching logic won't be in play unless the object passed in is a `re.Pattern` object, not a `str`.

Also, this minimal form should be safe from the many-tiny-queries problem that `DatasetRef` suffers from, at least for our current manager implementations, because those aggressively fetch all `DatasetTypes` from the database and stuff them in dict, so no DB calls should be needed here.  Hopefully that sort of aggressive fetching and/or caching scales up to the full science-user database...

Reply:
I have no idea why I was using `query` and not `get` there...